### PR TITLE
Add support for Tornado 6, increase test coverage

### PR DESCRIPTION
**Pull request by rklingsberg**

### CODE REVIEW

File: requires/installation.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+tornado>=4.0,<6.1
```
Comment (andrewrabert) [inline requires/installation.txt]: I'd prefer pinning to `<7` unless we find breakage in a minor release. Doing so would also be consistent another of other sprockets (sprockets.http, sprockets.mixins.metrics, etc).

I also realize that Tornado<5 is being kept around due to your PR only being a patch release.

File: tests.py [hunk 2]
```diff
@@ -35,3 +43,83 @@ def test_that_correlation_id_is_copied_from_request(self):
...
+        return web.Application([
+            (r'/status/(?P<status_code>\d+)', CorrelatedRequestHandler),
+            (r'/status/no-correlation/(?P<status_code>\d+)', RequestHandler),
+        ], log_function=correlation_id_logger)
+
+    def setUp(self):
+        self.patcher = unittest.mock.patch(
+            'sprockets.mixins.correlation.mixins.log.access_log')
+        self.access_logger = self.patcher.start()
+        super(CorrelationIDLoggerTests, self).setUp()
```
Comment (andrewrabert) [inline tests.py]: Remove redundant `super()` arguments. Python < 3 was dropped already.

File: tests.py [hunk 3]
```diff
@@ -35,3 +43,83 @@ def test_that_correlation_id_is_copied_from_request(self):
...
+
+    def tearDown(self):
+        self.patcher.stop()
+        super(CorrelationIDLoggerTests, self).tearDown()
```
Comment (andrewrabert) [inline tests.py]: Remove redundant `super()` args.

File: requires/development.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+-r requires/installation.txt
```
Comment (andrewrabert) [inline requires/development.txt:1]: Use `-e .` for dev reqs. Amongst other things that don't really apply to this project, it ensures deps as-seen-by `setup.py` are used. It's already being done in `.travis.yml` too.

File: sprockets/mixins/correlation/mixins.py [hunk 5]
```diff
@@ -46,7 +46,7 @@ def __init__(self, *args, **kwargs):
         # one exists.  We also want to set it in the outgoing response
         # which the property setter does for us.
         maybe_future = super(HandlerMixin, self).prepare()
-        if concurrent.is_future(maybe_future):
+        if concurrent.is_future(maybe_future):  # pragma: nocover
```
Comment (andrewrabert) [inline sprockets/mixins/correlation/mixins.py]: Don't ignore this as it's fairly critical functionality. A hole in coverage is preferable for now.

**Review by andrewrabert — CHANGES_REQUESTED** [review]

File: requires/installation.txt (hunk 1 above)
Comment (andrewrabert) [reply requires/installation.txt]: On second thought, this module is already coded using the `async` and `await` keywords. I'm pretty sure some of the earlier Tornado 4.x versions don't work correctly with those.

Comment (rklingsberg) [reply requires/installation.txt]: http://www.tornadoweb.org/en/stable/releases/v4.3.0.html#nov-6-2015

4.3 and up is good, changing pin to reflect that.

File: tests.py (hunk 2 above)
Comment (rklingsberg) [reply tests.py]: Done

File: tests.py (hunk 3 above)
Comment (rklingsberg) [reply tests.py]: Done

File: requires/development.txt (hunk 4 above)
Comment (rklingsberg) [reply requires/development.txt:1]: Added

File: sprockets/mixins/correlation/mixins.py (hunk 5 above)
Comment (rklingsberg) [reply sprockets/mixins/correlation/mixins.py]: Removed

**Review by andrewrabert — APPROVED** [review]

**Pull request merged**